### PR TITLE
Fixed histc return type for CUDA

### PR DESCRIPTION
**Pull request by izdeby**
Fixing reported [issue](https://github.com/pytorch/pytorch/issues/20208).

### CODE REVIEW

File: aten/src/ATen/native/cuda/SummaryOps.cu [hunk 1]
```diff
@@ -322,7 +322,7 @@ Tensor _histc_cuda_template(
     minvalue = minvalue - 1;
     maxvalue = maxvalue + 1;
   }
-  auto ret = cuda::CUDA_tensor_histogram<int64_t, input_t, false>(
+  auto ret = cuda::CUDA_tensor_histogram<float, input_t, false>(
```
Comment (zou3519) [inline aten/src/ATen/native/cuda/SummaryOps.cu]: The output dtype should probably match the input dtype. That is what I would expect, but you should also check it against the behavior of CPU histc.

**Review by zou3519 — CHANGES_REQUESTED** [review]

File: test/test_torch.py [hunk 2]
```diff
@@ -2458,58 +2458,58 @@ def test_zeros_out(self):
...
-            torch.tensor([0, 2, 1, 0], dtype=torch.double, device=device),
-            actual)
-        # mixed input
-        actual = torch.histc(
-            torch.tensor([1., 2, 1], dtype=torch.float, device=device),
-            bins=4, min=0, max=3)
-        self.assertEqual(
-            torch.tensor([0, 2, 1, 0], dtype=torch.float, device=device),
-            actual)
+    def test_histc(self):
```
Comment (zou3519) [inline test/test_torch.py]: You should probably delete the test in `test/test_cuda.py` now that you've moved tests for CUDA into this function.

File: test/test_torch.py [hunk 3]
```diff
@@ -2458,58 +2458,58 @@ def test_zeros_out(self):
...
+            with self.assertRaisesRegex(RuntimeError, 'bins must be > 0'):
+                torch.histc(torch.tensor([1], dtype=torch.float, device=device), bins=-1)
+
+            # without nbins
+            actual = torch.histc(
+                torch.tensor([2, 5], dtype=torch.float, device=device))
+            expected = torch.zeros(100, dtype=torch.float, device=device)
+            expected.data[0] = 1
+            expected.data[99] = 1
+            self.assertEqual(expected, actual)
```
Comment (zou3519) [inline test/test_torch.py]: I'm confused: the original code tested CUDA tensors by sending them through this path. Does `self.assertEqual` ignore dtype when comparing tensors?

Comment (izdeby) [reply test/test_torch.py]: apparently it does. i just tried this: 
`self.assertEqual(torch.tensor([1], dtype=torch.int), torch.tensor([1], dtype=torch.double))` and it didn't fail. I will create an issue for this cause its really concerning.

Comment (zou3519) [reply test/test_torch.py]: Thanks for checking that, that is concerning.

Could you add another test to check if the dtype of the output is correct, then?

**Review by zou3519 — APPROVED** [review]
LGTM, thank you

**Review by VitalyFedyunin — APPROVED** [review]

**Pull request closed without merging**